modeshape-jcr/src/main/java/org/modeshape/jcr/index/local/ManagedLocalIndex.java

```diff
@@ -85,4 +93,17 @@ public void removeAll() {
         index.removeAll();
     }
 
+    @Override
+    public IndexManager.IndexStatus getStatus() {
+        return status.get();
+    }
+
+    @Override
+    public void updateStatus(IndexManager.IndexStatus currentStatus, IndexManager.IndexStatus newStatus) {
+        this.status.compareAndSet(currentStatus, newStatus);
```

Comment:
nothing :smile: 
The reason I'm using `compareAndSet` and not `set` is that I expect to only be able to move between certain statuses in a predefined order. This method could return a `boolean`, but callers wouldn't be using it